spec/Section 4 -- Composition.md

```diff
@@ -3179,6 +3179,125 @@ input BookFilter {
 In this invalid case, `title` is mandatory in Schema A but not defined in
 `Schema B`, causing inconsistency in required fields across schemas.
 
+#### Required Argument Missing
+
+**Error Code:**  
+`REQUIRED_ARGUMENT_MISSING_IN_SOME_SCHEMA`
+
+**Severity:**  
+ERROR
+
+**Formal Specification:**
+
+- Let {typeNames} be the set of all object and interface types names from all
+  source schemas that are not declared as `@internal`
+- For each {typeName} in {typeNames}:
+  - Let {typeDefinitions} be the list of all type definitions from different
+    source schemas with the name {typeName}.
+  - Let {fieldNames} be the set of all field names from all {typeDefinitions}
+    that are not declared as `@internal`.
```

Comment:
Should they be `@shareable`?

BTW, is there a rule to prevent `@internal` and `@shareable` from being used at the same time?

Reply:
that' should be it's own rule i guess